packages/core/src/internal/newTask.js

```diff
@@ -88,12 +94,10 @@ export default function newTask(env, mainTask, parentContext, parentEffectId, me
 
     deferredEnd = deferred()
 
-    if (!running) {
-      if (aborted) {
-        deferredEnd.reject(taskError)
-      } else {
-        deferredEnd.resolve(taskResult)
-      }
+    if (status === ABORTED) {
+      deferredEnd.reject(taskError)
+    } else if (status === DONE) {
```

Comment:
CANCELLED should be resolved somehow too - in the Promise worls it seems like it should be treated as rejection (unfortunately) and that we should inject some artificial cancellation error here

Reply:
To resolve with TASK_CANCEL is okay for me.  Because when we join a task inside a try-block, the code would not jump into the catch-block if the task is cancelled, while an error would let the iterator jump to catch-block.

Reply:
seems reasonable 👌

Reply:
ok for me